app/controllers/admin/teachers_controller.rb

```diff
@@ -4,11 +4,20 @@ class TeachersController < ResourceController
     def create
       super do |teacher|
         teacher.password = teacher.password_confirmation = SecureRandom.hex
+        teacher.roles << Role.find(admin_id) if admin_role?
       end
     end
 
     protected
 
+    def admin_role?
+      params.dig("teacher", "role_ids").to_i == 1
+    end
```

Comment:
The check_box returns`0` for false / unchecked box and `1` for true / checked - which led to this implementation, unsure if the simple_form associations you suggested would provide something different (but I assume not).